newsfragments/46415.significant.rst

```diff
@@ -0,0 +1,16 @@
+Legacy signature for operator link is removed.
+
+``BaseOperatorLink.get_link`` used to accept execution date as an argument. This
+has been changed to accept ``ti_key`` to identify a task instance instead. The
+old signature, supported at runtime for compatibility, has been removed.
+
+* Types of change
+
+  * [x] Dag changes
+  * [ ] Config changes
+  * [ ] API changes
+  * [ ] CLI changes
+  * [ ] Behaviour changes
+  * [ ] Plugin changes
+  * [ ] Dependency changes
+  * [x] Code interface changes
```

Comment:
```suggestion
  * [x] Code interface changes

* Migration rules needed

  * ruff

    * AIR302
    
      * [ ] Signature of ``airflow.models.baseoperatorlink.BaseOperatorLink.get_link`` changed
      .. detailed in https://github.com/apache/airflow/pull/46415#issuecomment-2636186625
 
```

Reply:
Will we move it to task_sdk. I though we're to avoid users access to `airflow.models`

Reply:
I think so, althought that should be a separate change and not strictly related to this PR.

Does this fit in the AIR302 rule?

Reply:
> I think so, althought that should be a separate change and not strictly related to this PR.

agree. just want to confirm it. Thanks!

> Does this fit in the AIR302 rule?

we group any removal (but not moved to providers) into AIR302 now so I guess this could be an AIR302 one as well (dttm is removed and is replaced with something else)